### PR TITLE
Datr/add session host health check result

### PR DESCRIPTION
**Pull request by datr93**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json [hunk 1]
```diff
@@ -0,0 +1,4479 @@
...
+          "description": "The timestamp of the last update.",
+          "readOnly": true
+        },
+        "updateErrorMessage": {
+          "type": "string",
+          "description": "The error message."
+        },
+        "sessionHostHealthCheckResults": {
+          "type": "array",
+          "readOnly": true,
```
[the remark this answers is not in the document]
Comment (datr93) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json:3876]: ok thanks made it all read only

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json [hunk 2]
```diff
@@ -0,0 +1,4479 @@
...
+    },
+    "SessionHostHealthCheckReport": {
+      "type": "object",
+      "description": "The report for session host information.",
+      "properties": {
+        "healthCheckName": {
+          "type": "string",
+          "description": "Represents the name of the health check operation performed."
+        },
+        "healthCheckResult": {
```
[the remark this answers is not in the document]
Comment (datr93) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json:3893]: its flexible coming from another role. So I'll keep it as string

**Comment by datr93** [comment]
@jorgecotillo can you take another look address the comments!

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json [hunk 3]
```diff
@@ -3870,6 +3870,58 @@
...
+          "description": "Failure message: hints on what is wrong and how to recover.",
+          "readOnly": true
+        },
+        "errorCode": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Error code corresponding for the failure.",
+          "readOnly": true
+        },
+        "lastHealthCheckInUTC": {
```
Comment (RamyasreeChakka) [inline specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json:3920]: Please use property name 'lastHealthCheckDateTime'. Refer naming guidelines here: https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#176-date-and-time-properties

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json [hunk 4]
```diff
@@ -3870,6 +3870,58 @@
         "updateErrorMessage": {
           "type": "string",
           "description": "The error message."
+        },
+        "sessionHostHealthCheckResults": {
```
Comment (RamyasreeChakka) [inline specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json:3874]: What is the typical size of this array? Do you think all items would fit in a single request?

Comment (datr93) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json:3874]: we have around 10 health checks on average 400 bytes each. Although the list can grow. We think the size is still acceptable for a request.

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json (hunk 3 above)
Comment (datr93) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json:3920]: ok

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/examples/SessionHost_Delete.json [hunk 5]
```diff
@@ -26,7 +26,18 @@
...
-        "resourceId": "/subscriptions/daefabc0-95b4-48b3-b645-8a753a63c4fa/resourceGroups/resourceGroup1/providers/Microsoft.Compute/virtualMachines/sessionHost1"
+        "resourceId": "/subscriptions/daefabc0-95b4-48b3-b645-8a753a63c4fa/resourceGroups/resourceGroup1/providers/Microsoft.Compute/virtualMachines/sessionHost1",
+        "sessionHostHealthCheckResults": [
+          {
+            "healthCheckName": "0",
+            "healthCheckResult": "1",
+            "additionalFailureDetails": {
+              "message": "SessionHost healthy: is joined to domain ≤wvdarmtest1.net≥",
+              "errorCode": 0,
+              "lastHealthCheckInUTC": "2021-01-14T02:09:37.6236843Z"
```
Comment (jorgecotillo) [inline specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/examples/SessionHost_Delete.json]: Once you make the property name change, make sure to update your examples as well.

Comment (datr93) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/examples/SessionHost_Delete.json]: all updated

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (datr93) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json:3893]: correction this is an enum

**Comment by datr93** [comment]
@RamyasreeChakka @jorgecotillo can you guys take another look? and is one of you guys from the arm review board?

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (jorgecotillo) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json:3910]: If this is an enum, consider updating your property.

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json (hunk 3 above)
Comment (jorgecotillo) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json]: I still see the same property name (lastHealthCheckInUTC and not the recommended one).

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json (hunk 4 above)
Comment (jorgecotillo) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json:3874]: If the list can grow, then does it makes sense to add pagination?

Comment (datr93) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-01-14-preview/desktopvirtualization.json:3874]: i talked to the ack end team. They said it won't grow much. We can address if it comes. We control list size internally.

**Review by ChenTanyi — APPROVED** [review]

**Comment by ChenTanyi** [comment]
@datr93 please let me know when it is ready to merge.

**Comment by datr93** [comment]
@ChenTanyi  please merge!!!!

**Pull request merged**